### DISCUSSION TITLE
How to specify metadata via the command line?

### DISCUSSION THREAD

**Discussion by connorp** [quarto-dev/quarto-cli#2741] [Q&A] [answered]
I am trying to specify minimal output for the HTML format _via the command line_ and I haven't been able to figure out how to do it. Is it possible to set format-specific parameters via the command line? 

(My real goal is just to get HTML output without embedded CSS, but also without specifying a CSS file (I just want plain HTML), which I can do with pandoc directly on the command line via `-M document-css=false`, but I'm not sure how to get this result with Quarto. Any suggestions?

**Comment by jimjam-slam** [comment, marked as answer]
You'd probably wanna start with `theme: none`, which turns off Bootstrap! Not sure if there's any other CSS left after that, though 🤔

**Reply by connorp** [reply]
How would I do that on the command line? Neither `-M theme:none` and `-P theme:none` were effective for me.

**Reply by jimjam-slam** [reply]
Oooh, yeah. Well, normally you'd specify it under `format` and `html`:

```yaml
format:
  html:
    theme: none
```

But, for example, `quarto preview test.qmd -M 'format:\n  html:\n    theme:darkly'`, does not change the theme for me (and neither does simply `-M theme:darkly`). I'm not sure if there's a way to pass nested metadata over the command line!

**Reply by jimjam-slam** [reply]
I wonder if a [project profile](https://quarto.org/docs/projects/profiles.html) could pull this together? It's a bit messy, but 🤷🏻‍♂️

**Reply by connorp** [reply]
So I think the move here is to pass an additional YAML file with those contents, via `--execute-params`. However, that command line argument doesn't appear to work…? Probably time to file a different question or a bug report.

**Reply by mcanouil** [reply]
`--execute-params` is for parameters (see <https://quarto.org/docs/computations/parameters.html>) not metadata.

In order to change/update metadata, you need to use `--metadata`. 😉

**Reply by connorp** [reply]
Ah. Well it looks like `--metadata` only accepts `KEY:VALUE` arguments, rather than a passed YAML file, no? Is there a way to pass it a specified YAML file containing metadata? (That's why I jumped to `--execute-params`, it was the only argument in the man pages that referenced passing a YAML file).

**Reply by mcanouil** [reply]
Based on the output of `quarto render --help`I do not think there is a way to pass a yaml file for metadata, but Quarto accepts Pandoc command lines arguments.
So you could in theory use `--metadata-file` (<https://pandoc.org/MANUAL.html#option--metadata-file>).
